### PR TITLE
Added XSL transform to update codelist location

### PR DESCRIPTION
**Pull request by sachitrajbhandari**
https://github.com/aodn/backlog/issues/1568

### CODE REVIEW

**Comment by jonescc** [comment]
Should always update codelists - doesn't need to be an option

**Comment by sachitrajbhandari** [comment]
> Should always update codelists - doesn't need to be an option

Removed an option to update codelists.

**Comment by jonescc** [comment]
Unit tests need updating?

**Comment by jonescc** [comment]
Its also taking forever to run - we should probably look at compiling the stylesheets only once http://www.saxonica.com/documentation/#!using-xsl/compiling .  Not necessarily as a part of this backlog item - but it will hopefully speed things up significantly.

**Comment by sachitrajbhandari** [comment]
> Unit tests need updating?

Added Unit tests for codelist location update transform

File: geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java [hunk 1]
```diff
@@ -61,7 +66,7 @@ public void testAllMetadataTags() throws IOException {
             "TransformCatalogue",
             "-d", testDir.toString(),
             "-i", "metadata.xml",
-            "-o", "metadata.iso19115-3.2018.xml",
+            "-o", "2d2b2f92-12fa-4330-a480-94f0892c2b72.xml",
```
Comment (jonescc) [inline geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java:69]: Looks like we've hardcoded the test to a particular test case

Comment (jonescc) [reply geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java:69]: Only codeListLocation test runs successfully now.   Doesn't make sense to use a specific uuid for the output here unless thats the uuid input

Comment (sachitrajbhandari) [reply geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java:69]: Fixed. Removed hardcoded uuid.

Comment (jonescc) [reply geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java:69]: Unit tests still failing.

Comment (jonescc) [reply geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java:69]: Hardcoded uuid still remains

File: geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java [hunk 2]
```diff
@@ -71,7 +76,7 @@ public void testAllMetadataTags() throws IOException {
         for (File testCaseDir: testDir.toFile().listFiles(File::isDirectory)) {
             String testCaseName = testCaseDir.getName();
             Path expectedFile = resourceDir.resolve("expected").resolve(testCaseName + ".xml");
-            Path actualFile = testCaseDir.toPath().resolve("metadata/metadata.iso19115-3.2018.xml");
+            Path actualFile = testCaseDir.toPath().resolve("metadata/2d2b2f92-12fa-4330-a480-94f0892c2b72.xml");
```
Comment (jonescc) [inline geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java:79]: So this is wierd - I pyulled down latest changes to the branch locally - and this one has been fixed.  github issues?

File: geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java (hunk 1 above)
Comment (sachitrajbhandari) [reply geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java:69]: > Unit tests still failing.

Fixed. Updated Expected files with updated codelist locations.

Comment (jonescc) [reply geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java:69]: Still using specific uuid

File: geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java (hunk 2 above)
Comment (jonescc) [reply geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java:79]: And here

Comment (sachitrajbhandari) [reply geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java]: Fixed. Removed hardcoded uuid

File: geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java (hunk 1 above)
Comment (sachitrajbhandari) [reply geonetwork/transform-mcp-iso19115-3/src/test/java/au/org/aodn/xslt/TransformCatalogueTest.java]: Fixed. Removed specific uuid.

**Comment by jonescc** [comment]
Sorry Sachit - merge conflicts now.

**Comment by sachitrajbhandari** [comment]
> Sorry Sachit - merge conflicts now.

Resolved.

**Comment by jonescc** [comment]
Can't run tests - compilation error

**Comment by sachitrajbhandari** [comment]
> Can't run tests - compilation error

Fixed.

**Comment by jonescc** [comment]
OK.  So apparenttly I ran the unit tests against Alen's branch.   When I run the unit tests inb master after merging these changes only 3 unit tests pass.

Can you fix issues in master @sachitrajbhandari

**Comment by sachitrajbhandari** [comment]
> OK. So apparenttly I ran the unit tests against Alen's branch. When I run the unit tests inb master after merging these changes only 3 unit tests pass.
> 
> Can you fix issues in master @sachitrajbhandari

@Apaulic, could you please check. Following xml files have wrong uuid (d3feff71-ebe1-4b66-91c6-149beceef205)

https://github.com/aodn/utilities/blob/master/geonetwork/transform-mcp-iso19115-3/src/test/resources/dataParameters/expected/7d3a2249-6f9a-4dc9-8bbb-354009f5b254.xml
https://github.com/aodn/utilities/blob/master/geonetwork/transform-mcp-iso19115-3/src/test/resources/dataParameters/export/7d3a2249-6f9a-4dc9-8bbb-354009f5b254/metadata/metadata.xml

https://github.com/aodn/utilities/blob/master/geonetwork/transform-mcp-iso19115-3/src/test/resources/aggregationInfo/expected/10729bf5-fadb-49e5-a775-4d2546946fe1.xml
https://github.com/aodn/utilities/blob/master/geonetwork/transform-mcp-iso19115-3/src/test/resources/aggregationInfo/export/10729bf5-fadb-49e5-a775-4d2546946fe1/metadata/metadata.xml

**Pull request merged**